Tools/clinic/clinic.py

```diff
@@ -748,10 +756,10 @@ class CLanguage(Language):
     stop_line     = "[{dsl_name} start generated code]*/"
     checksum_line = "/*[{dsl_name} end generated code: {arguments}]*/"
 
-    def __init__(self, filename):
+    def __init__(self, filename: str) -> None:
         super().__init__(filename)
         self.cpp = cpp.Monitor(filename)
-        self.cpp.fail = fail
+        setattr(self.cpp, "fail", fail)
```

Comment:
Here again, I'd just leave it as it is and tell mypy to go away:

```suggestion
        self.cpp.fail = fail  # type: ignore[method-assign]
```

Mypy doesn't really understand what we're doing either way, so better just to keep the code idiomatic Python imo